protocol/chainlib/chainproxy/rpcclient/http.go

```diff
@@ -200,7 +200,7 @@ func (hc *httpConn) doRequest(ctx context.Context, msg interface{}) (io.ReadClos
 	if err != nil {
 		return nil, err
 	}
-	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
```

Comment:
I wanted to validate the HandleStatusError here for tendermint rpc + jsonrpc. we get it here for free. add a method to the client expose it to the error handler and reuse it there

Reply:
because now we dont check 429 and 504 for tendermint and jsonrpc RPC clients.